src/main/java/it/geosolutions/httpproxy/Utils.java

```diff
@@ -139,4 +145,18 @@ static final String getProxyHostAndPort(ProxyInfo proxyInfo) {
             return null;
         }
     }
+
+    static URL buildURL(String value) throws MalformedURLException {
+        URL url = new URL(value);
+
+        if(url.getPort() == -1) {
+            if(url.toString().startsWith("https://")) {
```

Comment:
I would change this to url.getProtocol().equals("https")

Reply:
done